File: 04 - JavaScript/08-04-00-Paso-Por-Referencia-ULP.js
// Funciones en JavaScript
// En JavaScript los objetos y los arreglos se pasan por valor
// Pero el valor que se pasa es la referencia del objeto o arreglo
function cambiarValor(parametro){
    parametro[0] = 20
}

// Llamamos a la funcion
let argumento = [10]
console.log(`Antes de llamar funcion: ${argumento}`)
cambiarValor(argumento)
console.log(`Despues de llamar funcion: ${argumento}`)

File: 03 - C++/08-04-00-Paso-Por-Referencia-ULP.cpp
#include <iostream>
using namespace std;

// Cambiar valor (paso por referencia)
void cambiarValor(int *parametro){
	*parametro = 20;
}

int main() {
	// Ejemplo paso por referencia
	int argumento = 10;
	cout << "Antes llamar funcion: " << argumento << endl;
	cambiarValor( &argumento );
	cout << "Despues llamar funcion: " << argumento;
	
	return 0;
}

